depinject/provider_desc.go

```diff
@@ -2,21 +2,24 @@ package depinject
 
 import (
 	"reflect"
+	"strings"
+	"unicode"
 
 	"github.com/pkg/errors"
+	"golang.org/x/exp/slices"
 )
 
-// ProviderDescriptor defines a special provider type that is defined by
+// providerDescriptor defines a special provider type that is defined by
 // reflection. It should be passed as a value to the Provide function.
 // Ex:
 //
-//	option.Provide(ProviderDescriptor{ ... })
-type ProviderDescriptor struct {
+//	option.Provide(providerDescriptor{ ... })
+type providerDescriptor struct {
```

Comment:
This is what is failing the build: https://github.com/cosmos/cosmos-sdk/blob/main/core/internal/registry.go#L24-L25

Reply:
Yes there was a replace directive but when I remove it go can't resolve the tagged module